app/models/job_invocation.rb

```diff
@@ -108,4 +110,16 @@ def output(host)
     return unless (task = sub_task_for_host(host)) && task.main_action && task.main_action.live_output.any?
     task.main_action.live_output.first['output']
   end
+
+  def generate_description!
+    key_re = /%\{([^\}]+)\}/
+    template_invocation = template_invocations.first
+    input_names = template_invocation.template.template_input_names
+    hash_base = Hash.new { |hash, key| hash[key] = "%{#{key}}" }
+    input_hash = hash_base.merge Hash[input_names.zip(template_invocation.input_values.pluck(:value))].symbolize_keys
+    input_hash.update(:job_name => job_name)
+    description_format.scan(key_re) { |key| input_hash[key.first.to_sym] }
```

Comment:
Calling `to_sym` on user input has security consequences (DOS due to symbols not getting garbage collected). What about replacing missnig keys in description_format with "%%{missing_key}", the symbolize keys has a similar issues (the set of inputs is an user input as well). It seems the easies thing would be to implement the '%{}' substitution using `gsub` or similar.